lib/HLSL/HLMatrixBitcastLowerPass.cpp

```diff
@@ -193,31 +194,42 @@ void MatrixBitcastLowerPass::lowerMatrix(Instruction *M, Value *A) {
         SmallVector<Value *, 2> idxList(GEP->idx_begin(), GEP->idx_end());
         DXASSERT(idxList.size() == 2,
                  "else not one dim matrix array index to matrix");
-
-        HLMatrixType MatTy = HLMatrixType::cast(EltTy);
-        Value *matSize = Builder.getInt32(MatTy.getNumElements());
-        idxList.back() = Builder.CreateMul(idxList.back(), matSize);
+        if (!DM.GetShaderModel()->IsSM69Plus()) {
```

Comment:
Can we base the low-level decisions on a descriptively named boolean field in the `MatrixBitcastLowerPass` class, initialized based on the shader model, instead of passing the DxilModule down and checking the shader model each time?

Reply:
Maybe even something about scalarizing operations, rather than anything to do with the native DXIL vector support?